src/main/java/ca/uwaterloo/drinkmasterapi/feature/mqtt/controller/MqttClientController.java

```diff
@@ -0,0 +1,53 @@
+package ca.uwaterloo.drinkmasterapi.feature.mqtt.controller;
+
+import ca.uwaterloo.drinkmasterapi.feature.mqtt.service.MqttClientService;
+import com.amazonaws.services.iot.client.AWSIotException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.UnsupportedEncodingException;
+import java.security.NoSuchAlgorithmException;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/mqtt")
+public class MqttClientController {
+
+    @Autowired
+    private MqttClientService mqttClientServiceImpl;
+
+    @GetMapping("/gdpr")
+    public ResponseEntity publishGdprMessage(@RequestParam("locationId") final String locationId) throws AWSIotException {
+        this.mqttClientServiceImpl.publishGdprMessage(locationId);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping("/ota")
+    public ResponseEntity publishOtaMessage(@RequestParam("locationId") final String locationId,
+                                            @RequestParam("url") final String url,
+                                            @RequestParam("productIndex") final String productIndex) throws AWSIotException, UnsupportedEncodingException, NoSuchAlgorithmException {
+        this.mqttClientServiceImpl.publishOtaMessage(locationId, url, productIndex);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping("/label")
+    public ResponseEntity publishLabelMessage(@RequestParam("locationId") final String locationId,
+                                              @RequestParam("baseUrl") final String baseUrl,
+                                              @RequestParam("productIndex") final String productIndex) throws AWSIotException {
+        this.mqttClientServiceImpl.publishLabelMessage(locationId, baseUrl, productIndex);
+        return ResponseEntity.ok().build();
+    }
+
+    //todo: add drinkmaster endpoint
+    @GetMapping("/pour")
```

Comment:
I think all these endpoints should be post requests, and create a post request class called `PourMessageRequest` or somthing

Reply:
it s mqtt req, not related to database, it doesn't matter to use post or get